contracts/src/river/registry/facets/stream/StreamRegistry.sol

```diff
@@ -114,6 +114,11 @@ contract StreamRegistry is IStreamRegistry, RegistryModifiers {
       revert(RiverRegistryErrors.BAD_ARG);
     }
 
+    // Delete genesis miniblock if `stream` still contains the genesis block after `stream` has advanced since genesis.
+    if (stream.lastMiniblockNum == 0) {
+      delete ds.genesisMiniblockByStreamId[streamId];
+    }
```

Comment:
```suggestion
    if (ds.genesisMiniblockByStreamId[streamId].length != 0) {
      delete ds.genesisMiniblockByStreamId[streamId];
    }
```
How about this or always delete? Need to test the gas difference though.
```solidity
    delete ds.genesisMiniblockByStreamId[streamId];
```

Reply:
If one of these options save on gas we should go for that ofc. But I don't see why either the length comparison or the always delete option would be more efficient. But that is maybe my lack of solidity/evm knowledge. Could you do the simulations and make the change if one of these options is more efficient? Not sure what a good way is to test the differences.

Reply:
I think simply
```solidity
    delete ds.genesisMiniblockByStreamId[streamId];
```
is the most efficient. What the bytecode generated by the compiler does is reading the length of the bytes array and erasing the storage slots one by one in a loop. If the length of the mini block is zero, it will skip.
I will do benchmarks when I work on `StreamRegistry` later. But this should do it for now. It will also remove the genesis mini block for streams whose `lastMiniblockNum` is no longer 0.